computer_vision/haralick_descriptors.py

```diff
@@ -0,0 +1,433 @@
+            base_pixel = image[x, y]
+            offset_pixel = image[x + offset_x, y + offset_y]
+
+            matrix[base_pixel, offset_pixel] += 1
+    matrix_sum = np.sum(matrix)
+    return matrix / (1 if matrix_sum == 0 else matrix_sum)
+
+
+def haralick_descriptors(matrix: np.ndarray) -> list:
+    """Calculates all 8 Haralick descriptors based on co-occurence input matrix.
+    All descriptors are as follows:
+    Maximum probability, Inverse Difference, Homogeneity, Entropy,
+    Energy, Dissimilarity, Contrast and Correlation
+
+    Args:
+        matrix: Co-occurence matrix to use as base for calculating descriptors.
+
+    Returns:
+        Reverse ordered list of resulting descriptors
+
+    Example:
+        >>> img = np.array([[[108, 201, 72], [255, 11,  127]],
+        ...                 [[56,  56,  56], [128, 255, 107]]])
+        >>> gray = grayscale(img)
+        >>> binary = binarize(gray)
+        >>> morphological = opening_filter(binary)
+        >>> mask_1 = binary_mask(gray, morphological)[0]
+        >>> concurrency = matrix_concurrency(mask_1, [0, 1])
+        >>> haralick_descriptors(concurrency)
+        [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    """
+    # Function np.indices could be used for bigger input types,
+    # but np.ogrid works just fine
+    i, j = np.ogrid[0 : matrix.shape[0], 0 : matrix.shape[1]]  # np.indices()
+
+    # Pre-calculate frequent multiplication and subtraction
+    prod = np.multiply(i, j)
+    sub = np.subtract(i, j)
+
+    # Calculate numerical value of Maximum Probability
+    maximum_prob = np.max(matrix)
+    # Using the definition for each descriptor individually to calculate its matrix
+    correlation = prod * matrix
+    energy = np.power(matrix, 2)
+    contrast = matrix * np.power(sub, 2)
+
+    dissimilarity = matrix * np.abs(sub)
+    inverse_difference = matrix / (1 + np.abs(sub))
+    homogeneity = matrix / (1 + np.power(sub, 2))
+    entropy = -(matrix[matrix > 0] * np.log(matrix[matrix > 0]))
+
+    # Sum values for descriptors ranging from the first one to the last,
+    # as all are their respective origin matrix and not the resulting value yet.
+    return [
+        maximum_prob,
+        correlation.sum(),
+        energy.sum(),
+        contrast.sum(),
+        dissimilarity.sum(),
+        inverse_difference.sum(),
+        homogeneity.sum(),
+        entropy.sum(),
+    ]
+
+
+def get_descriptors(
+    masks: tuple[np.ndarray, np.ndarray], coordinate: list
+) -> np.ndarray:
+    """
+    Calculate all Haralick descriptors for a sequence of
+    different co-occurrence matrices, given input masks and coordinates.
+
+    Example:
+        >>> img = np.array([[[108, 201, 72], [255, 11,  127]],
+        ...                 [[56,  56,  56], [128, 255, 107]]])
+        >>> gray = grayscale(img)
+        >>> binary = binarize(gray)
+        >>> morphological = opening_filter(binary)
+        >>> get_descriptors(binary_mask(gray, morphological), [0, 1])
```

Comment:
```suggestion
        >>> get_descriptors(binary_mask(gray, morphological), (0, 1))
```
Change list to tuple (see previous comments)